WMF Framework/Remote Notifications/RemoteNotificationsController.swift

```diff
@@ -1,180 +1,334 @@
-        operationsController = RemoteNotificationsOperationsController(session: session, configuration: configuration, languageLinkController: languageLinkController, authManager: authManager)
+        
+        self.apiController = RemoteNotificationsAPIController(session: session, configuration: configuration)
         self.configuration = configuration
         self.authManager = authManager
+        self.languageLinkController = languageLinkController
+        
         super.init()
         
+        do {
+            modelController = try RemoteNotificationsModelController()
+        } catch (let error) {
+            DDLogError("Failed to initialize RemoteNotificationsModelController: \(error)")
+            modelController = nil
+        }
+        
         NotificationCenter.default.addObserver(self, selector: #selector(applicationDidBecomeActive), name: UIApplication.didBecomeActiveNotification, object: nil)
         NotificationCenter.default.addObserver(self, selector: #selector(authManagerDidLogIn), name:WMFAuthenticationManager.didLogInNotification, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(authManagerDidLogOut), name: WMFAuthenticationManager.didLogOutNotification, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(modelControllerDidLoadPersistentStores(_:)), name: RemoteNotificationsModelController.didLoadPersistentStoresNotification, object: nil)
+        
     }
     
-    @objc private func applicationDidBecomeActive() {
-        refreshNotifications(force: false) { _ in
-            //
+    // MARK: NSNotification Listeners
+    
+    @objc private func modelControllerDidLoadPersistentStores(_ note: Notification) {
+        
+        guard let modelController = modelController else {
+            return
         }
+        
+        if let object = note.object, let error = object as? Error {
+            DDLogError("RemoteNotificationsModelController failed to load persistent stores with error \(error); not instantiating RemoteNotificationsOperationsController")
+            return
+        }
+        
+        operationsController = RemoteNotificationsOperationsController(languageLinkController: languageLinkController, authManager: authManager, apiController: apiController, modelController: modelController)
+        
+        populateFilterStateFromPersistence()
     }
     
-    @objc private func authManagerDidLogIn() {
-        importNotificationsIfNeeded { _ in
-            //
-        }
+    @objc private func applicationDidBecomeActive() {
+        loadNotifications(force: false)
     }
     
-    @objc func deleteLegacyDatabaseFiles() {
+    @objc private func authManagerDidLogOut() {
         do {
-            try operationsController.deleteLegacyDatabaseFiles()
+            try modelController?.resetDatabaseAndSharedCache()
```

Comment:
This is venturing into totally fine if you'd rather we address it in a separate PR territory, but I think to be good data citizens we should also reset the `filterState` and `allInboxProjects` in `RemoteNotificationsController` when the user logs out. If we don't, the previous `filterState` is retained, and if they decide to log back in their first Notifications Center experience is in an unexpected state. It might be a little tricky because of `filterState`'s `didSet` property observer – it may be a little wasteful and silly and not proper, but maybe even this little bit in the listening for log out method is sufficient:

```swift 
@objc private func authManagerDidLogOut() {
    do {
	filterState = RemoteNotificationsFilterState(readStatus: .all, offTypes: [], offProjects: [])
        allInboxProjects = []
        try modelController?.resetDatabaseAndSharedCache()
    }
	...
}
```

Reply:
Great catch! I agree this is needed. I think your approach works well - it crossed my mind that maybe we should allow `filterState` to be nullable but then we'll have to handle the optional everywhere. We may be persisting some unnecessary data but I'm okay with that tradeoff at this point. Plus I'm pretty sure we already are persisting this default elsewhere, if they reset everything on the filters & inbox views.